eng/GenerateAnalyzerNuspec.targets

```diff
@@ -86,8 +86,16 @@
       <AnalyzerConfigurationFileDir>$(RepoRoot)docs</AnalyzerConfigurationFileDir>
       <AnalyzerConfigurationFileName>Analyzer Configuration.md</AnalyzerConfigurationFileName>
     </PropertyGroup>
-    
-    <MSBuild Projects="$(RepoRoot)src\Tools\GenerateDocumentationAndConfigFiles\GenerateDocumentationAndConfigFiles.csproj" Targets="Build">
+
+    <PropertyGroup>
+      <!--
+        Source build's runtime environment is incomplete, so we call through a stub project to force assembly resolution
+        to choose versions from the available subset.
+      -->
+      <GenerateDocumentationAndConfigFilesProjectSuffix Condition="'$(DotNetBuildFromSource)' == 'true'">ForBrokenRuntime</GenerateDocumentationAndConfigFilesProjectSuffix>
```

Comment:
Is `ForBrokenRuntime` the best description of this? Perhaps we want to indicate that this is related to source build? Also, what is broken? Can we file a bug?

Reply:
I already renamed this once. It's not inaccurate, but perhaps could be better.

> Also, what is broken?

Source build omits some stable public releases of some dependencies in violation of the backwards compatibility expectations of those projects. From the perspective of those projects, this is a broken runtime design.

> Can we file a bug?

We should file a feature request for allowing a different set of build/runtime dependencies without creating a new project: https://github.com/dotnet/roslyn-analyzers/pull/4965#discussion_r598843475